hydragnn/preprocess/raw_dataset_loader.py

```diff
@@ -202,20 +288,22 @@ def __scale_features_by_num_nodes(self, dataset):
         ]
 
         for idx, data_object in enumerate(dataset):
-            dataset[idx].y[scaled_graph_feature_index] = (
-                dataset[idx].y[scaled_graph_feature_index] / data_object.num_nodes
-            )
-            dataset[idx].x[:, scaled_node_feature_index] = (
-                dataset[idx].x[:, scaled_node_feature_index] / data_object.num_nodes
-            )
+            if dataset[idx].y is not None:
+                dataset[idx].y[scaled_graph_feature_index] = (
+                    dataset[idx].y[scaled_graph_feature_index] / data_object.num_nodes
+                )
+            if dataset[idx].x is not None:
+                dataset[idx].x[:, scaled_node_feature_index] = (
+                    dataset[idx].x[:, scaled_node_feature_index] / data_object.num_nodes
+                )
 
         return dataset
 
     def __normalize_dataset(self):
 
         """Performs the normalization on Data objects and returns the normalized dataset."""
         num_node_features = self.dataset_list[0][0].x.shape[1]
```

Comment:
Similar thing with `node_feature_dim` like the line below

Reply:
If there are no graph features, then` dataset[idx].y` is not defined. And accessing it would make the code crash in that case.

Reply:
I was proposing this change for line 217
```suggestion
        num_node_features = self.node_feature_dim
```

Reply:
I am always a bit uncertain on this. `self.node_feature_dim` is retrieved by the `config` file.
maybe we can check somewhere, through an assert, that  `self.dataset_list[0][0].x.shape[1] == self.node_feature_dim`

Reply:
Is this resolved?